content/blog/2022-05-04-cloudwatch-metric-alarms/index.md

```diff
@@ -0,0 +1,234 @@
+
+Here are some basic commands to get started:
+```sh
+# add new metric data
+$ awslocal cloudwatch put-metric-data \
+   --namespace test \
+   --metric-data '[{"MetricName": "Orders", "Value": 20}]' 
+
+# get information about all collected metrics
+$ awslocal cloudwatch list-metrics
+{
+    "Metrics": [
+        {
+            "Namespace": "test",
+            "MetricName": "Orders",
+            "Dimensions": []
+        }
+    ]
+}
+
+# query metric data
+$ awslocal cloudwatch get-metric-data \
+   --metric-data-queries '[{"Id": "my-id","MetricStat": {"Metric": {"Namespace": "test","MetricName": "Orders"},"Period": 3600, "Stat": "Sum" }}]' \
+   --start-time 2022-05-04T08:00:00Z \
+   --end-time  2022-05-04T19:00:00Z
+{
+    "MetricDataResults": [
+        {
+            "Id": "my-id",
+            "Label": "Orders Sum",
+            "Timestamps": [
+                "2022-05-04T09:00:00Z"
+            ],
+            "Values": [
+                20.0
+            ],
+            "StatusCode": "Complete"
+        }
+    ]
+}
+```
+You can also use our **LocalStack WebApp** and navigate to [https://app.localstack.cloud/resources/monitoring](https://app.localstack.cloud/resources/monitoring). Here you can add metrics (and alarms) as well, supported by a nice UI, which already gives you configurable parameters.
+
+
+In order to create an alarm, you need to define what kind of metrics the alarm should evaluate. 
+LocalStack supports metric alarms with simple statics. This typically includes:
+* **metric-name**
+* **namespace**
+* **threshold** - a double value
+* **comparison-operator** - e.g. _GreaterThanThreshold_, _LessThanThreshold_, _GreaterThanOrEqualToThreshold_, _LessThanOrEqualToThreshold_
+* **statistic** - e.g. _SampleCount_, _Average_, _Sum_, _Minimum_, _Maximum_
+* **period** - the interval in seconds used to calculate the metric
+* **evaluation-periods** - defines the number of periods that will be evaluated to check the threshold
+* **treat-missing-data** - defines how missing data points should be evaluated: _missing_, _ignore_, _notBreaching_, _breaching_
+
+For a deeper understanding of alarm evaluation, we advice to consult the official [AWS docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/AlarmThatSendsEmail.html#alarm-evaluation).
+
+Even though LocalStack's implementation of CloudWatch is not yet feature complete, you can already cover a range of use-cases. Next, we look into how we can make use of CloudWatch alarms with Lambdas.
+
+## Setup a Lambda Function
+For the sake of simplicity, we will create a Lambda that will always fail. This will make it easier to demonstrate the alarms functionality. If you are interested in further learning about Lambdas, you can check our previous blog post on [Hot Swapping Python Lambda Functions using LocalStack](../2022-03-07-hot-swapping-python-lambda-functions-using-localstack), which also gives a solid introduction to Lambdas.
```

Comment:
```suggestion
For simplicity, we will create a Lambda that will always fail. This will make it easier to demonstrate the alarm's functionality. Suppose you are interested in further learning about Lambdas. In that case, you can check our previous blog post on [Hot Swapping Python Lambda Functions using LocalStack](../2022-03-07-hot-swapping-python-lambda-functions-using-localstack), which also gives a solid introduction to Lambdas.
```